### PR TITLE
New report formats

### LINKED ISSUE #8: HTML or direct PDF output
I was really excited to try this out, but then really bummed by only having XML or LaTeX output - I can't read either. I spent like 20 minutes trying to set up a Tex distribution and going through the loop of compiling, failing, installing another Tex style, and repeating until I gave up because of some weird error about the fontawesome style.

My point is, it would be awesome if flan output some easily human readable format directly, without the need for some external dependencies to process it (no Tex, no XML parser, etc.). For now, I'm sorta stuck squinting at these XML files....

### LINKED ISSUE #13: Markdown report support
For the output the LaTeX file are not that usual, and can not be show in the repo in a human readable format.

I believe the best format for the report's output's is a Makrdown because it can be shown in the Github ,or something else, in a easier way.

Also, form what I saw the output is simple and only have colorful boxes for the severity. These can be handler by the Markdown file

### LINKED ISSUE #44: Produce the report in HTML, not LaTeX.
I spent 30 minutes installing various LaTeX packages and extra fonts on Ubuntu 18.04 to be able to view the LaTeX report the scan generates.

Reports in LaTeX format is a usability bug.

Just generate report in plain HTML.

### PR DESCRIPTION
**Pull request by shadowscatcher**
HTML, Markdown and JSON report formats.
resolves #8, resolves #13, resolves #44 
Output format can be passed as file extension using enviroment variable "format". If not passed, Latex will be used.

### CODE REVIEW

File: contrib/report_builders/html_report_builder.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+from contrib.descriptions import VulnDescriptionProvider
+from contrib.report_builders import JsonReportBuilder
+
+
+class JinjaHtmlReportBuilder(JsonReportBuilder):
+    def __init__(self, description_provider: VulnDescriptionProvider):
+        super().__init__(description_provider)
+        self.template_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'templates')
+        self.template_name = 'jinja2_report.html'
```
Comment (nadinelyab) [inline contrib/report_builders/html_report_builder.py]: I tested this and got the following error. Looks like jinja2_report.html doesnt exist?
```
Traceback (most recent call last):
  File "/output_report.py", line 81, in <module>
    main(*sys.argv[1:4], report_type=report_format)
  File "/output_report.py", line 64, in main
    builder = create_report_builder(report_type)
  File "/output_report.py", line 58, in create_report_builder
    return builder_map[report_type](provider)
  File "/output_report.py", line 50, in <lambda>
    'html': lambda p: JinjaHtmlReportBuilder(p),
  File "/contrib/report_builders/html_report_builder.py", line 15, in __init__
    self._read_template()  # type: Template
  File "/contrib/report_builders/html_report_builder.py", line 23, in _read_template
    self._template = template_env.get_template(self.template_name)
  File "/usr/local/lib/python3.5/site-packages/jinja2/environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/usr/local/lib/python3.5/site-packages/jinja2/environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "/usr/local/lib/python3.5/site-packages/jinja2/loaders.py", line 113, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "/usr/local/lib/python3.5/site-packages/jinja2/loaders.py", line 187, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: jinja2_report.html
```

File: contrib/report_builders/report_builder.py [hunk 2]
```diff
@@ -8,80 +8,62 @@
 
 
 class ReportBuilder(metaclass=abc.ABCMeta):
-    @abc.abstractmethod
```
Comment (nadinelyab) [inline contrib/report_builders/report_builder.py]: Why did you decide to remove these?

File: contrib/report_builders/html_report_builder.py (hunk 1 above)
Comment (shadowscatcher) [reply contrib/report_builders/html_report_builder.py]: Yes, my bad, I forgot to add template file to vcs

File: contrib/report_builders/report_builder.py (hunk 2 above)
Comment (shadowscatcher) [reply contrib/report_builders/report_builder.py]: Those decorators obligate type to implement all methods, however, some report builder types (both json and html) do not need them. Most methods only useful if report internaly is built in sequential order, in that case, they still can be used, but it is not imperative now.

File: Makefile [hunk 3]
```diff
@@ -3,4 +3,13 @@ build :
 
 container_name = flan_$(shell date +'%s')
 start : 
-	docker run --name $(container_name) -v "$(pwd)/shared:/shared:Z" flan_scan
+	docker run --name $(container_name) -v "$(shell pwd)/shared:/shared:Z" flan_scan
```
[the remark this answers is not in the document]
Comment (shadowscatcher) [reply Makefile]: Done for all recipes.

File: contrib/report_builders/templates/jinja2_report.html [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+      margin-left: -25px;
+    }
+  </style>
+</head>
+<body>
+<h1>Flan scan report</h1>
+<h2>{{ data.start_date }}</h2>
+<h2>Summary</h2>
+<div id="summary_review">
+    Flan Scan ran a network vulnerability scan with the following Nmap command on Sun Nov 24 20:20:15 2019:
```
Comment (nadinelyab) [inline contrib/report_builders/templates/jinja2_report.html]: The date should not be hardcoded here

Comment (shadowscatcher) [reply contrib/report_builders/templates/jinja2_report.html]: Good point.

**Comment by nadinelyab** [comment]
Looks good! I like that the html format closely resembles the Latex format. Thanks for your contributions!

**Pull request merged**